File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int numer1, int denom1, int numer2, int denom2) {
    int* answer = (int*)malloc(1);
    int denom = denom1 * denom2;
    int numer = (numer1 * denom2) + (numer2 * denom1);
    int gcd = 1;
    for(int i = 1; i <= numer || i <= denom; i++){
        if(numer % i == 0 && denom % i == 0){
            gcd= i;
        }
    }
    answer[0] = numer/gcd;
    answer[1] = denom/gcd;
    return answer;
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.kt
class Solution {
    fun solution(numer1: Int, denom1: Int, numer2: Int, denom2: Int): IntArray {        
        var top = (numer1 * denom2) + (numer2 * denom1)
        var down = denom1 * denom2
        
        val num = gcd(top, down)
        
        return intArrayOf(top / num, down / num)
    }
    
    fun gcd(a: Int, b: Int): Int{
        if (b == 0){
            return a
        }
        return gcd(b, a % b)
    }
}